cpp/ql/src/semmle/code/cpp/exprs/Access.qll

```diff
@@ -198,20 +262,18 @@ private predicate referenceConversion(Conversion c) {
 /**
  * Holds if `e` is a reference expression (that is, it has a type of the
  * form `T&`), which is converted to a value. For example:
- *
  * ```
  * int myfcn(MyStruct &x) {
  *   return x.field;
  * }
  * ```
- *
  * In this example, the type of `x` is `MyStruct&`, but it gets implicitly
  * converted to `MyStruct` in the expression `x.field`.
  */
 private predicate exprHasReferenceConversion(Expr e) { referenceConversion(e.getConversion+()) }
 
 /**
- * A field access of a field of `this`. The access has no qualifier because
+ * A field access of a field of `this` which has no qualifier because
  * the use of `this` is implicit. For example, `field` is equivalent to
  * `this->field` if `field` is a member of `this`.
  *
```

Comment:
I haven't given an example for `ImplicitThisFieldAccess` because I'm not sure I can write something that clears up, rather than adds to, the confusion between this and a `PointerFieldAccess` with a compiler generated `this` qualifier.  I don't know which situations would give you one or the other.  Perhaps even `ImplicitThisFieldAccess` should just be rolled into `PointerFieldAccess`, with a note that in this situation you may get no qualifier???

Reply:
I discovered that you can sometimes divine such things by grepping the `test` sub-tree for the QL class in question, which will lead you to the concrete syntax example(s).

Reply:
I've dug deeper using the tests like you suggested, and made my best attempt at an explained example.